Contributions/Linux_Memory_Management_Essentials.md

```diff
@@ -0,0 +1,249 @@
+# **Linux Memory Management Essentials (Work In Progress)**
+
+## Index
+
+[Terms and Abbreviations](#Terms-and-Abbreviations)
+
+[References](#References)
+
+[Disclaimer](#Disclaimer)
+
+[Purpose of the document](#Purpose-of-the-document)
+
+[Structure of the document](#Structure-of-the-document)
+
+[Kernel-space memory allocations](Kernel-space-memory-allocations)
+
+[User-space memory allocations](User-space-memory-allocations)
+
+[License: CC BY-SA 4.0](#License-CC-BY-SA-40)
+
+
+## **Terms and Abbreviations**
+Plese refer to the Linux Kernel documentation.
+   
+
+## **References**
+
+1. [Linux Kernel website](https://www.kernel.org) - <https://www.kernel.org>
+2. ***Interference Scenarios for an ARM64 Linux System***
+3. [CC BY-SA 4.0 Deed | Attribution-ShareAlike 4.0 International | Creative Commons](https://creativecommons.org/licenses/by-sa/4.0/) - <https://creativecommons.org/licenses/by-sa/4.0/> License
+
+
+## **Disclaimer**
+* This document is not intended to be a replacement for understanding the memory management of the Linux Kernel,
+  nor it attempts to be an exhaustive analysis of safety implications.
+* Because of the very volatile nature of the code within the Linux Kernel, each of the statements made
+  below should not be taken at face value, but rather verified, for any Linux kernel version following
+  the one used while writing the document (6.9).
+* When referring to specific HW features, the document refers to the ARM64 architecture.
+
+## **Purpose of the document**
+This document aims to provide an holistic view of what happens in Linux memory management, so that
+one is at least aware of certain features and can use this document as a jumping pad toward more detailed documentation.
+Or even to the code base.
+
+## **Structure of the document**
+The document is divided in two parts, based on the destinatary of the memory allocations discussed: kernel-space or user-space.
+Individual points are numbered for ease of reference, but the numbering is not meant to represent any sequence.
+
+## **Memory management in Linux**
+
+### **Kernel-space memory allocations**
+
+#### **Directly Verifiable Assertions**
+The following section presents a set of statements that can be objectively verified e.g. by inspecting the sources.
+
+1. unlike processes memory, kernel memory pages are not swapped, nor dropped silently by the kernel itself,
```

Comment:
should we add that kernel memory swapping can happen during PM Suspend ?

Reply:
I'd rather say here "dynamically swapped" and then have a separate comment about suspend, which is a valid topic but happens under different conditions.

Reply:
I would just add a note saying that we are not considering any swapping or any other memory management mechanism involved along PM routines....

Reply:
from the perspective of a newcomer it might be useful to add maybe more words - I'll think about some compromise

Reply:
I agree that power management considerations could be a separate point, but we could perhaps make the following changes to clarify this point: 

```suggestion
1. Unlike processes memory, kernel memory pages are not dynamically swapped during normal (i.e. not low-power) operation, nor dropped silently by the kernel itself,
```

The point about hypervisors could then be extended to note that memory may be swapped when entering or leaving low-power states. It might be better to break this out as a separate sentence.